### PR TITLE
Move all VCS plugins to separate Gradle projects

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: plugins/package-configuration-providers/ort-config/src/main/kotlin/OrtConfigPackageConfigurationProvider.kt [hunk 1]
```diff
@@ -23,7 +23,7 @@ import java.io.File
 
 import org.apache.logging.log4j.kotlin.logger
 
-import org.ossreviewtoolkit.downloader.vcs.Git
+import org.ossreviewtoolkit.downloader.VersionControlSystem
```
Comment (mnonnenmacher) [inline plugins/package-configuration-providers/ort-config/src/main/kotlin/OrtConfigPackageConfigurationProvider.kt:26]: Commit message: "based **on**"

File: downloader/src/main/kotlin/vcs/GitWorkingTree.kt [hunk 2]
```diff
@@ -50,7 +50,7 @@ private fun findGitOrSubmoduleDir(workingDirOrFile: File): Repository {
     }.getOrThrow()
 }
 
-open class GitWorkingTree(
+internal open class GitWorkingTree(
```
Comment (mnonnenmacher) [inline downloader/src/main/kotlin/vcs/GitWorkingTree.kt:53]: Commit message: "part**l**y"

**Review by mnonnenmacher — CHANGES_REQUESTED** [review]

File: downloader/src/main/kotlin/vcs/MercurialWorkingTree.kt [hunk 3]
```diff
@@ -25,7 +25,7 @@ import org.ossreviewtoolkit.downloader.WorkingTree
 import org.ossreviewtoolkit.model.VcsType
 import org.ossreviewtoolkit.utils.common.ProcessCapture
 
-class MercurialWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(workingDir, vcsType) {
+internal class MercurialWorkingTree(workingDir: File, vcsType: VcsType) : WorkingTree(workingDir, vcsType) {
```
Comment (fviernau) [inline downloader/src/main/kotlin/vcs/MercurialWorkingTree.kt:28]: Should the commit title indicate that it's a "breaking" change?

**Review by fviernau — APPROVED** [review]

File: downloader/src/main/kotlin/vcs/MercurialWorkingTree.kt (hunk 3 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/MercurialWorkingTree.kt:28]: Strictly speaking yes, otherwise @nnobelis will complain again 😉 I'll change it.

Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/MercurialWorkingTree.kt:28]: Ok, wow, I just realized that GitHub now keep your approval even thou I did not only rebase, but also reword a commit message. So GitHub realizes that the actual code in the PR did not change, this is nice!

**Review by fviernau — APPROVED** [review]

**Pull request merged**